parsing-scripts/create_maintainers_fixtures.py

```diff
@@ -0,0 +1,103 @@
+
+        try:
+            github = maintainer['github']
+            github_provided = True
+        except KeyError:
+            github_provided = False
+
+        maintainer_object = {}  # Define empty maintainer Object
+        github_repeat = False
+        email_repeat = False
+
+        # If this is first entry, add to the list
+        if temp_maintainers == []:  # Making the First Entry
+            print('reached empty once')
+            if github_provided:
+                maintainer_object['github'] = github
+            if email_provided:
+                maintainer_object['name'] = name
+                maintainer_object['domain'] = domain
+
+        # If not first entry, check for uniqueness
+        j = 0
+        for check in temp_maintainers:
+            if email_provided:
+                try:
+                    if check['name'] == name and check['domain'] == domain:
+                        email_repeated_at = j
+                        email_repeat = True
+                except KeyError:
+                    pass
+            if github_provided:
+                try:
+                    if check['github'] == github:
+                        github_repeat = True
+                        github_repeated_at = j
+                except KeyError:
+                    pass
+            j = j + 1
+            if github_repeat and email_repeat:
+                break
+
+        # First Case, neither GitHub not email is repeated
+        if github_repeat is False and email_repeat is False and temp_maintainers != []:
+            if github_provided:
+                maintainer_object['github'] = github
+            if email_provided:
+                maintainer_object['name'] = name
+                maintainer_object['domain'] = domain
+
+        # Second Case, both are repeated at the same place ( can be skipped )
+        elif github_repeat and email_repeat and github_repeated_at == email_repeated_at:
+            continue
+
+        # Third Case, Github is repeated but not the provided email ( same github handle has different emails )
+        elif github_repeat and email_repeat is False and email_provided:
+            maintainer_object['github'] = github
+            maintainer_object['inconsistency'] = True
+            maintainer_object['name'] = name
+            maintainer_object['domain'] = domain
+
+        # Fourth Case, Email is repeated but not the provided GitHub handle ( create new entry )
+        elif email_repeat and github_repeat is False and github_provided:
+            maintainer_object['name'] = name
+            maintainer_object['domain'] = domain
+            maintainer_object['github'] = github
+            maintainer_object['inconsistency'] = True
+
+        # Fifth Case, Github is repeated and Email is not provided (skip)
+        elif github_repeat and email_provided is False:
+            continue
+
+        # Sixth Case, Email is repeated and Github is not provided (skip)
```

Comment:
This whole file sounds a lot more complicate than it could be if my first idea was implemented. Maybe this needs further discussion.

Reply:
This is almost the same idea you suggested in the email. The further complications have arrived just because not all maintainers have provided their GitHub handles. Otherwise it would have been very simple.